src/process.cr

```diff
@@ -285,9 +285,13 @@ class Process
     writer_pipe.close
     bytes = uninitialized UInt8[4]
     if reader_pipe.read(bytes.to_slice) == 4
-      reader_pipe.close
       errno = IO::ByteFormat::SystemEndian.decode(Int32, bytes.to_slice)
```

Comment:
Can we clean this up? Passing errno and message size (both Int32) must have the same logic. IMO a simple `io.write_bytes(x)` and `x = io.read_bytes(Int32)` is enough. No need to deal with endianess in parent/child IPC —it should be on *both* sides anyway.

Reply:
This is done because there's no `read_bytes?(Int32)` which does not raise on EOF. EOF is the hot path, the case where the exec succeeds. So the *first* read call has to be coded like this check for EOF without having the overhead of raising and rescuing an exception in the successful case. Adding `read_bytes?(Int32)` which allows you to check for EOF would be nice, but it's another PR.